File: 1_iniciante/javascript/1021.js
/**
 * @source Beecrowd
 * @problem 1021 - Banknotes and Coins
 * @subject Sequential
 *
 * @author Letícia Cruz
 * @date 2020-10-15
 */

var input = require("fs").readFileSync("/dev/stdin", "utf8");
var lines = input.split("\n");

let valor = parseFloat(lines.shift());

let cem = parseInt(valor / 100);
let cinquenta = parseInt((valor % 100) / 50);
let vinte = parseInt(((valor % 100) % 50) / 20);
let dez = parseInt((((valor % 100) % 50) % 20) / 10);
let cinco = parseInt(((((valor % 100) % 50) % 20) % 10) / 5);
let dois = parseInt((((((valor % 100) % 50) % 20) % 10) % 5) / 2);
let um = parseInt((((((valor % 100) % 50) % 20) % 10) % 5) % 2);

let decimal = valor - parseInt(valor);
decimal = parseInt(decimal * 100);

let mcinquenta = parseInt(decimal / 50);
let mvinte = parseInt((decimal % 50) / 25);
let mdez = parseInt(((decimal % 50) % 25) / 10);
let mcinco = parseInt((((decimal % 50) % 25) % 10) / 5);
let mum = parseInt((((decimal % 50) % 25) % 10) % 5);

console.log("NOTAS:");
console.log(cem + " nota(s) de R$ 100.00");
console.log(cinquenta + " nota(s) de R$ 50.00");
console.log(vinte + " nota(s) de R$ 20.00");
console.log(dez + " nota(s) de R$ 10.00");
console.log(cinco + " nota(s) de R$ 5.00");
console.log(dois + " nota(s) de R$ 2.00");
console.log("MOEDAS:");
console.log(um + " moeda(s) de R$ 1.00");
console.log(mcinquenta + " moeda(s) de R$ 0.50");
console.log(mvinte + " moeda(s) de R$ 0.25");
console.log(mdez + " moeda(s) de R$ 0.10");
console.log(mcinco + " moeda(s) de R$ 0.05");
console.log(mum + " moeda(s) de R$ 0.01");

File: 1_iniciante/java/1021.java
/**
 * @source Beecrowd
 * @problem 1021 - Banknotes and Coins
 * @subject Sequential
 * 
 * @author Letícia Cruz
 * @date 2025-04-21
 */

import java.io.IOException;
import java.util.Scanner;

class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        int[] tiposNotas = {2, 5, 10, 20, 50, 100};
        int[] tiposMoedas = {1, 5, 10, 25, 50, 100};

        double valor = sc.nextDouble();

        System.out.println("NOTAS:");

        int i;
        int inteiroContado = 0;
        int notaI;
        double notas;
        for (i = tiposNotas.length - 1; i >= 0; i--) {
            notaI = tiposNotas[i];
            notas = (double) notaI;
            int qtdNotasI = (int) ((valor - inteiroContado) / notaI);
            System.out.printf("%d nota(s) de R$ %.2f\n", qtdNotasI, notas);

            inteiroContado += qtdNotasI * notaI;
        }

        int moedaI;
        int decimal = (int) ((valor - inteiroContado) * 100);
        int decimalContado = 0;
        double moeda;

        System.out.println("MOEDAS:");
        for (i = tiposMoedas.length - 1; i >= 0; i--) {
            moedaI = tiposMoedas[i];
            int qtdMoedasI = (decimal - decimalContado) / moedaI;
            moeda = (double) moedaI;
            moeda = moeda / 100;
            System.out.printf("%d moeda(s) de R$ %.2f\n", qtdMoedasI, moeda);

            decimalContado += qtdMoedasI * moedaI;
        }

        sc.close();
    }

}
